src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php

```diff
@@ -249,7 +249,7 @@ public function supportsSavepoints()
      */
     public function getDateFormat()
     {
-        return 'Y-m-d H:i:s.000';
+        return 'Y-m-d H:i:s.u';
```

Comment:
You need to have the field type datetime2

https://github.com/laravel/framework/blob/adf7abc0b72766fa804cd6bc5e6772110f09a3e4/src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php#L477

needs to be datetime2(3)

https://github.com/laravel/framework/blob/adf7abc0b72766fa804cd6bc5e6772110f09a3e4/src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php#L522
https://github.com/laravel/framework/blob/adf7abc0b72766fa804cd6bc5e6772110f09a3e4/src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php#L542
ditto